src/reducers/systemIntakesReducer.ts

```diff
@@ -1,19 +1,35 @@
-import { STORE_SYSTEM_INTAKES } from '../constants/actions';
-import { SystemIntakesState } from '../types/systemIntake';
+import { SystemIntakesState } from 'types/systemIntake';
+import { prepareSystemIntakeForApp } from 'data/systemIntake';
+import { fetchSystemIntakes } from 'types/routines';
+import { DateTime } from 'luxon';
+import { Action } from 'redux-actions';
 
 const initialState: SystemIntakesState = {
   systemIntakes: []
 };
 
 function systemIntakesReducer(
   state = initialState,
-  action: any
+  action: Action<any>
 ): SystemIntakesState {
   switch (action.type) {
-    case STORE_SYSTEM_INTAKES:
+    case fetchSystemIntakes.REQUEST:
       return {
         ...state,
-        systemIntakes: action.systemIntakes
+        isLoading: true
+      };
+    case fetchSystemIntakes.SUCCESS:
+      return {
+        ...state,
+        systemIntakes: action.payload.map((intake: any) =>
+          prepareSystemIntakeForApp(intake)
+        )
+      };
+    case fetchSystemIntakes.FULFILL:
+      return {
+        ...state,
+        isLoading: false,
+        loadedTimestamp: DateTime.local()
```

Comment:
I'm thinking maybe `loadedTimestamp` should get set on `SUCCESS`. I'm not sure it should set the timestamp if the request is a failure. This is a minor technicality because I don't think the timestamp is rendered anywhere unless there is data from `SUCCESS`.

Reply:
If there's a failure, we'd want to put the error message and display it.

Reply:
I thought about the error too, and realized that we don't have a use for it yet. I really need to finish the doc of potential errors for design.